usr/lib/common/asn1.c

```diff
@@ -1741,7 +1739,15 @@ CK_RV ber_encode_RSAPublicKey(CK_BBOOL length_only, CK_BYTE **data,
     /* need a bitstring */
     ber = ber_alloc_t(LBER_USE_DER);
     rc = ber_put_bitstring(ber, (char *)buf2, len * 8, 0x03);
-    rc = ber_flatten(ber, &val);
+    rc |= ber_flatten(ber, &val);
```

Comment:
Well, either `ber_flatten` works after a failing `ber_put_bitstring`, or it doesn't. It does not really matter, since rc is non-zero anyway, and it goes to error handling.  The only point is that the error handling should also free `val`, just in case `ber_flatten` did work, and return a non-NULL val. So I'll add a `ber_bvfree(val);` to the error handling for those places where it is not yet so, and make sure val is initialized with NULL.

Reply:
There are a couple of places where this is the case.

Reply:
Yes, I saw that. I guess I have covered them all now. Let me know if you find places that I missed.

Reply:
Arg, actually `ber_put_bitstring` returns the length of the ber encoding it created, so a check for rc = 0 is wrong. Need to fix this. Testcases are failing, but actually travis does not really check for testcase failures....

Reply:
Everything fixed now.